sanic/worker/inspector.py

```diff
@@ -25,111 +34,156 @@ def __init__(
-                except timeout:
-                    continue
-                else:
-                    action = conn.recv(64)
-                    if action == b"reload":
-                        self.reload()
-                    elif action == b"shutdown":
-                        self.shutdown()
-                    elif action.startswith(b"scale"):
-                        num_workers = int(action.split(b"=", 1)[-1])
-                        logger.info("Scaling to %s", num_workers)
-                        self.scale(num_workers)
-                    else:
-                        data = dumps(self.state_to_json())
-                        conn.send(data.encode())
-                    conn.send(b"\n")
-                    conn.close()
-        finally:
-            logger.info("Inspector closing")
-            sock.close()
-
-    def stop(self, *_):
-        self.run = False
-
-    def state_to_json(self):
+    def _state_to_json(self) -> Dict[str, Any]:
         output = {"info": self.app_info}
-        output["workers"] = self.make_safe(dict(self.worker_state))
+        output["workers"] = self._make_safe(dict(self.worker_state))
         return output
 
-    def reload(self):
+    @staticmethod
+    def _make_safe(obj: Dict[str, Any]) -> Dict[str, Any]:
+        for key, value in obj.items():
+            if isinstance(value, dict):
+                obj[key] = Inspector._make_safe(value)
+            elif isinstance(value, datetime):
+                obj[key] = value.isoformat()
+        return obj
+
+    def reload(self) -> None:
         message = "__ALL_PROCESSES__:"
         self._publisher.send(message)
 
-    def scale(self, num_workers: int):
+    def scale(self, replicas) -> str:
+        num_workers = 1
+        if replicas:
+            num_workers = int(replicas)
+        log_msg = f"Scaling to {num_workers}"
+        logger.info(log_msg)
         message = f"__SCALE__:{num_workers}"
         self._publisher.send(message)
+        return log_msg
 
-    def shutdown(self):
+    def shutdown(self) -> None:
         message = "__TERMINATE__"
         self._publisher.send(message)
 
-    @staticmethod
-    def make_safe(obj: Dict[str, Any]) -> Dict[str, Any]:
-        for key, value in obj.items():
-            if isinstance(value, dict):
-                obj[key] = Inspector.make_safe(value)
-            elif isinstance(value, datetime):
-                obj[key] = value.isoformat()
-        return obj
-
 
-def inspect(host: str, port: int, action: str):
-    out = sys.stdout.write
-    with socket(AF_INET, SOCK_STREAM) as sock:
-        try:
-            sock.connect((host, port))
-        except ConnectionRefusedError:
-            error_logger.error(
-                f"{Colors.RED}Could not connect to inspector at: "
-                f"{Colors.YELLOW}{(host, port)}{Colors.END}\n"
-                "Either the application is not running, or it did not start "
-                "an inspector instance."
+class InspectorClient:
```

Comment:
This class would probably be better in a module other than `worker`. Maybe `client` or similar.

Reply:
I will move it to `./cli`.